src/rust/ide/lib/view/src/graph_editor/builtin/visualization/native/documentation.rs

```diff
@@ -0,0 +1,184 @@
+/// Content in the documentation view when the data is yet to be received.
+const PLACEHOLDER_STR : &str = "<h3>Enso Documentation Viewer</h3><p>No documentation available</p>";
+
+
+/// Generates documentation view stylesheet.
+pub fn get_doc_style() -> String {
+    format!("<style>{}</style>", include_str!("documentation/style.css"))
+}
+
+#[derive(Clone,CloneRef,Debug)]
+#[allow(missing_docs)]
+pub struct ViewModel {
+    logger : Logger,
+    dom    : DomSymbol,
+    size   : Rc<Cell<Vector2>>,
+}
+
+impl ViewModel {
+    /// Constructor.
+    fn new(scene:&Scene) -> Self {
+        let logger          = Logger::new("DocumentationView");
+        let div             = web::create_div();
+        let dom             = DomSymbol::new(&div);
+        let screen          = scene.camera().screen();
+        let doc_view_height = screen.height - (DOC_VIEW_MARGIN * 2.0);
+        let size_vec        = Vector2(DOC_VIEW_WIDTH,doc_view_height);
+        let size            = Rc::new(Cell::new(size_vec));
+
+        dom.dom().set_style_or_warn("white-space"     ,"normal"                        ,&logger);
+        dom.dom().set_style_or_warn("overflow-y"      ,"auto"                          ,&logger);
+        dom.dom().set_style_or_warn("overflow-x"      ,"auto"                          ,&logger);
+        dom.dom().set_style_or_warn("background-color","rgba(255, 255, 255, 0.85)"     ,&logger);
+        dom.dom().set_style_or_warn("padding"         ,"5px"                           ,&logger);
+        dom.dom().set_style_or_warn("pointer-events"  ,"auto"                          ,&logger);
+        dom.dom().set_style_or_warn("border-radius"   ,format!("{}px", CORNER_RADIUS)  ,&logger);
+        dom.dom().set_style_or_warn("width"           ,format!("{}px", DOC_VIEW_WIDTH) ,&logger);
+        dom.dom().set_style_or_warn("height"          ,format!("{}px", doc_view_height),&logger);
+
+        scene.dom.layers.main.manage(&dom);
+        ViewModel {dom,logger,size}.init()
+    }
+
+    fn init(self) -> Self {
+        self.reload_style();
+        self
+    }
+
+    /// Sets size of the documentation view.
+    fn set_size(&self, size:Vector2) {
+        self.size.set(size);
+        self.reload_style();
+    }
+
+    /// Generates HTML documentation for documented suggestion.
+    fn gen_doc(doc: String) -> FallibleResult<String> {
+        let parser = parser::DocParser::new()?;
+        let output = parser.generate_html_doc_pure(doc);
+        Ok(output?)
+    }
+
+    fn receive_data(&self, data:&Data) -> Result<(),DataError> {
+        let data_inner = match data {
+            Data::Json {content} => content,
+            _                    => todo!(),
+        };
+
+        let data_str = serde_json::to_string_pretty(&**data_inner);
+        let data_str = data_str.unwrap_or_else(|e| format!("<Cannot render data: {}>", e));
+        // Fixes a Doc Parser Bug - to be removed when rewritten to rust
```

Comment:
I was writing about it in my previous PR - Im wondering why the comment is not here anymore. If you add such comments, you have to always paste in the comment link to an issue on board describing the error. Moreover, comments are sentences, should have dots at the end. Moreover, is this comment applied to 1 line or 2 lines? If 2 lines -refactor them to separate function and write fixme comment on top of the function